health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/components/editVillagePopulationPopUP.js

```
@@ -17,24 +19,50 @@ const EditVillagePopulationPopUp = ({ onClose, census }) => {
+  const mutation = Digit.Hooks.useCustomAPIMutationHook({
+    url: "/census-service/_update", // Replace with the appropriate API endpoint
+  });
 
-    // Close the popup after saving
-    onClose();
+  const handleSave = async () => {
+    // Determine workflow action based on user roles
+    let workflowAction = ""; // Default action
+
+    // Check user roles and set workflow action accordingly
+    if (userRoles && userRoles.includes('POPULATION_DATA_APPROVER')) {
```

Biome

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)